File: mocks/medl-mock/src/main/java/no/nav/pensjon/vtp/mocks/medl/medl2/rest/api/v1/MedlemskapsunntakMock.java
package no.nav.pensjon.vtp.mocks.medl.medl2.rest.api.v1;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiParam;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import javax.validation.constraints.NotNull;
import java.util.List;
import java.util.Set;

import static no.nav.pensjon.vtp.mocks.medl.medl2.rest.api.v1.MedlemskapsunntakApiParams.*;

@RestController
@RequestMapping(value = "medl2/api/v1/medlemskapsunntak", produces = MediaType.APPLICATION_JSON_VALUE)
@Api(tags = {"Medlemskapsunntak"})
public class MedlemskapsunntakMock {
    @GetMapping(value = "/{unntakId}")
    @ApiOperation(API_OPERATION_MEDLEMSKAPSUNNTAK)
    public Object hentMedlemskapsunntak(
            @ApiParam(API_PARAM_INKLUDER_SPORINGSINFO) @RequestParam("inkluderSporingsinfo") Boolean inkluderSporing,
            @ApiParam(value = API_PARAM_UNNTAK_ID, required = true) @NotNull @PathVariable("unntakId") Long unntakId) {
        return null;
    }

    @GetMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
    @ApiOperation(API_OPERATION_MEDLEMSKAPSUNNTAK_I_PERIODE)
    public List<Object> hentMedlemskapsunntakIPeriode(
            @ApiParam(API_PARAM_STATUSER) @RequestParam("statuser") Set<String> statuser,
            @ApiParam(API_PARAM_INKLUDER_SPORINGSINFO_PERSON) @RequestParam("inkluderSporingsinfo") boolean sporing) {

        return List.of();
    }
}

File: mocks/medl-mock/src/main/kotlin/no/nav/pensjon/vtp/mocks/medl/medl2/rest/api/v1/MedlemskapsunntakMock.kt
package no.nav.pensjon.vtp.mocks.medl.medl2.rest.api.v1

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import io.swagger.annotations.ApiParam
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import javax.validation.constraints.NotNull

@RestController
@RequestMapping(value = ["rest/medl2/api/v1/medlemskapsunntak"])
@Api(tags = ["Medlemskapsunntak"])
class MedlemskapsunntakMock {
    @GetMapping(value = ["/{unntakId}"])
    @ApiOperation("Henter ut informasjon om et spesifikt medlemskapsunntak.")
    fun hentMedlemskapsunntak(
        @ApiParam("Hvorvidt man er interessert i å se sporingsinformasjonen som er registrert på perioden.") @RequestParam(
            "inkluderSporingsinfo"
        ) inkluderSporing: Boolean?,
        @ApiParam(
            value = "Den funksjonelle ID'en til perioden man ønsker å hente.",
            required = true
        ) @PathVariable("unntakId") unntakId: @NotNull Long?
    ) = null

    @GetMapping
    @ApiOperation("Henter ut medlemskapsunntak for en spesifikk person.")
    fun hentMedlemskapsunntakIPeriode(
        @ApiParam("Dersom man ikke ønsker å hente perioder uansett status kan man spesifisere hvilke man er interessert i her.") @RequestParam(
            "statuser"
        ) statuser: Set<String?>?,
        @ApiParam("Hvorvidt man er interessert i å se sporingsinformasjonen som er registrert på periodene som hentes ut.") @RequestParam(
            "inkluderSporingsinfo"
        ) sporing: Boolean
    ): List<Any> = emptyList()
}
